File: backend/src/main/java/com/dev/backend/entity/Genre.java
package com.dev.backend.entity;

import java.util.List;

import com.dev.backend.constant.BaseStatus;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Entity
@Table(name = "genres")
public class Genre extends BaseEntity<Integer> {

    @Column(nullable = false)
    private String name;

    @Column(unique = true, nullable = false)
    private String slug;

    private String urlImage;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private BaseStatus status = BaseStatus.ACTIVE;

    @OneToMany(mappedBy = "genre")
    private List<ProductGenre> productGenres;
}

File: frontend/src/features/admin/genre/types/genre.ts
import type { BaseStatus } from "@/types/status";


export interface GenreResponse {
  id: number;
  name: string;
  status: BaseStatus;
  urlImage: string;
  totalProduct: number;
}


export interface GenreRequest {
  name: string;
  status: BaseStatus;
  previewImageUrl: string;
  
}



